dali/image/image_factory.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/image/image_factory.h"
+#include "dali/image/generic_image.h"
+#include "dali/image/png.h"
+#include "dali/image/bmp.h"
+#include "dali/image/jpeg.h"
+
+namespace dali {
+
+namespace {
+
+bool CheckIsJPEG(const uint8 *jpeg, int) {
```

Comment:
Yes, it should. However, there's a reason I did it this way.
The point of it is that the simple factory "pattern", which is here now, I would like to change in future to factory method. Then, the checks are valid in corresponding classes. However, it would require further redesigning and since I wanted to have small change, it looks this way for now. It will change in future

Reply:
Can we ask each Image class to register somehow with and we can iterate over all registered one instead of touching this file every time new image class appear.

Reply:
Result of discussion in person:
This can be achieved using preprocessor macros. Might be valid solution, but requires putting more thought in it. I'll mark it as TODO